### PR TITLE
Reschedule service for architect request

### PR DESCRIPTION
**Pull request by ahmedhamilastage**

### CODE REVIEW

File: app/architect_request/services/ArchitectRequestService.py [hunk 1]
```diff
@@ -313,3 +316,38 @@ def architect_request_paginated(cls, request):
...
+        Handles rescheduling an ArchitectRequest.
+
+        Args:
+            architect_request_id (int): The ID of the ArchitectRequest to be rescheduled.
+            data (dict): The validated data for rescheduling the ArchitectRequest.
+
+        Returns:
+            Response: The response object containing the result of the operation.
+        """
+        try:
```
Comment (MohamedMegdich1) [inline app/architect_request/services/ArchitectRequestService.py:332]: use one try

File: app/architect_request/routes/ArchitectRequestUrls.py [hunk 2]
```diff
@@ -54,4 +54,9 @@
         ArchitectRequestViewSet.as_view({"post": "add_note"}),
         name="architect-request-add-note",
     ),
+    path(
+        "reschedule/<int:pk>/",
```
Comment (MohamedMegdich1) [inline app/architect_request/routes/ArchitectRequestUrls.py:58]: change to reschedule-meeting

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: app/architect_request/controllers/ArchitectRequestViewSet.py [hunk 3]
```diff
@@ -179,3 +182,22 @@ def add_note(self, request, pk=None):
             Response: The response object containing the result of the operation.
         """
         return ArchitectRequestService.add_note_to_architect_request(pk, request.data)
+
+    @action(
+        detail=True,
+        methods=["PUT"],
+        url_path="reschedule",
```
Comment (MohamedMegdich1) [inline app/architect_request/controllers/ArchitectRequestViewSet.py:189]: change it also reschedule-meeting

**Review by MohamedMegdich1 — CHANGES_REQUESTED** [review]

File: app/architect_request/controllers/ArchitectRequestViewSet.py (hunk 3 above)
Comment (ahmedhamilastage) [reply app/architect_request/controllers/ArchitectRequestViewSet.py]: done

**Review by MohamedMegdich1 — APPROVED** [review]

**Pull request merged**